src/msg/async/compression_onwire.cc

```diff
@@ -0,0 +1,75 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "compression_onwire.h"
+#include "common/dout.h"
+
+#define dout_subsys ceph_subsys_ms
+
+ceph::compression::onwire::rxtx_t ceph::compression::onwire::rxtx_t::create_handler_pair(
+    CephContext* ctx, 
+    const CompConnectionMeta& comp_meta,
+    std::uint64_t compress_min_size) {
+    if (comp_meta.is_compress()) {
+        CompressorRef compressor = Compressor::create(ctx, comp_meta.get_method());
+
+        if (compressor != nullptr) {
+            return {std::make_unique<RxHandler>(ctx, compressor), 
+                    std::make_unique<TxHandler>(ctx, compressor, comp_meta.get_mode(), compress_min_size)};
+        }
+    } 
+    
+    return {nullptr, nullptr};
+}
+
+bool ceph::compression::onwire::TxHandler::compress(const ceph::bufferlist &input, ceph::bufferlist &out) {
+    boost::optional<int32_t> compressor_message;
+    if (m_init_onwire_size < m_min_size) {
+        return false;
+    }
+
+    m_compress_potential -= input.length();
```

Comment:
A good reason to use int64_t instead of uint64_t for m_compress_potential
(and for all variables holding quantities)